### PR TITLE
sql: fix mixed-version behaviour of create_tenant()

### PR DESCRIPTION
**Pull request by postamar**
Previously, running create_tenant on the latest binary in
a mixed-version cluster would bootstrap the system schema using the
logic from the latest binary, in which all system database migration
upgrade steps have been "baked in".

This might not be compatible with the the tenant cluster's version which
is (correctly) set to the active host cluster version, the idea there
being that tenant clusters versions cannot be greater than host cluster
versions.

This commit fixes this by version-gating the tenant cluster system
schema bootstrapping, and using hard-coded values when bootstrapping
into the old version.

Informs https://github.com/cockroachdb/cockroach/issues/94773.

Release note: None

### CODE REVIEW

**Review by fqazi — APPROVED** [review]
> Reviewed 5 of 5 files at r1, 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94774)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

**Review by rafiss — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94774)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @fqazi and @postamar)

___
*[`pkg/jobs/registry.go` line 533 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/94774#-NL8ScTDD1twFWeYklRH:-NL8ScTDD1twFWeYklRI:bw8o9w0) ([raw file](https://github.com/cockroachdb/cockroach/blob/aac5192217c483037e894ada69f5dc37e7c23bd6/pkg/jobs/registry.go#L533)):*
> ```Go
> 			ctx, "job-row-update", txn, updateStmt, jobID, jobType.String(),
> 		); err != nil {
> 			if pgerror.GetPGCode(err) != pgcode.UndefinedColumn {
> ```

consider using `select column_name from [show columns from system.jobs] where column_name = 'job_type'` rather than executing the query and checking for an error. it should be possible to use pg_catalog also 

or if we have the system.jobs descriptor, we could use the `FindColumnWithName` function
___
*[`pkg/upgrade/upgrades/permanent_upgrades.go` line 71 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/94774#-NL8TxWW0z_WtIyEL_gB:-NL8TxWW0z_WtIyEL_gC:b-gnp45m) ([raw file](https://github.com/cockroachdb/cockroach/blob/aac5192217c483037e894ada69f5dc37e7c23bd6/pkg/upgrade/upgrades/permanent_upgrades.go#L71)):*
> ```Go
> 		ctx, "addRootToAdminRole", nil /* txn */, updateMembership, username.AdminRole, username.RootUser, username.AdminRoleID, username.RootUserID)
> 	if err != nil {
> 		if pgerror.GetPGCode(err) == pgcode.UndefinedColumn {
> ```

similar to above, could use `show columns` here, or inspect the `system.role_members` descriptor

File: pkg/sql/catalog/bootstrap/previous_release.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+const (
+	// nonSystem222 is the hard-coded string representation of the
+	// initial values for a non-system tenant as produced by the 22.2
+	// release binary.
+	//
+	// This string literal was generated by copying over the expected results
+	// for the TestInitialValuesToString test in the release-22.2 branch.
+	nonSystem222 = `[{"key":""}
```
Comment (ajwerner) [inline pkg/sql/catalog/bootstrap/previous_release.go:62]: what's this first entry about?

**Review by ajwerner — APPROVED** [review]
I quite like this. @dt does it please you? What integration testing should accompany this?

File: pkg/sql/catalog/bootstrap/previous_release.go (hunk 1 above)
Comment (postamar) [reply pkg/sql/catalog/bootstrap/previous_release.go:62]: Keys without values are split keys and this one is the split we define at the tenant prefix, but in this datum the prefix is stripped away leaving the empty string.

**Comment by postamar** [comment]
Thanks for the reviews. I'm going to bors this now before it grows stale.

bors r+

**Pull request merged**